src/shims/unix/fd.rs

```diff
@@ -36,6 +36,30 @@ pub trait FileDescription: std::fmt::Debug + Any {
         throw_unsup_format!("cannot write to {}", self.name());
     }
 
+    /// Reads as much as possible into the given buffer from a given offset,
+    /// and returns the number of bytes read.
+    fn pread<'tcx>(
+        &mut self,
+        _communicate_allowed: bool,
+        _bytes: &mut [u8],
+        _offset: u64,
+        _ecx: &mut MiriInterpCx<'tcx>,
+    ) -> InterpResult<'tcx, io::Result<usize>> {
+        throw_unsup_format!("cannot pread from {}", self.name());
```

Comment:
We _specifically_ try to avoid this behavior. Miri is a cross-interpreter; we do everything reasonable to ensure that shims behave the same no matter what host is selected.

In this case, I think these shims can be implemented so that they work across targets by doing an explicit seek first; this is definitely slower but it is portable and should fulfill the required API.

Reply:
IIUC Windows simply does not have a similar syscall. This is why `FileExt` is gated on `unix` in the standard library. See this [discussion](https://internals.rust-lang.org/t/19649) for more information. It may be possible to emulate it with `Seek`, but it would require either locking, or file descriptor duplication.

Reply:
I would prefer for this PR to be merged as-is and to discuss Windows emulation in a separate PR/issue.